File: src/main/java/com/github/joelws/release/tracker/dao/ReleaseDaoImpl.java
package com.github.joelws.release.tracker.dao;

import com.github.joelws.release.tracker.common.GenericDaoImpl;
import com.github.joelws.release.tracker.model.Release;
import org.springframework.stereotype.Repository;

@Repository
public class ReleaseDaoImpl extends GenericDaoImpl<Release, String> implements ReleaseDao
{
    public ReleaseDaoImpl()
    {
        super(Release.class);
    }
}

File: src/main/kotlin/com/joelws/release/tracker/dao/release/ReleaseDaoImpl.kt
package com.joelws.release.tracker.dao.release

import com.joelws.release.tracker.dao.GenericDaoImpl
import com.joelws.release.tracker.entity.release.Release

import org.springframework.stereotype.Repository

@Repository
open class ReleaseDaoImpl : GenericDaoImpl<Release, String>(), ReleaseDao {

    override fun list(): List<Release> {
        val query = "SELECT r FROM Release r"
        @Suppress("UNCHECKED_CAST")
        return entityManager.createQuery(query).resultList as List<Release>
    }
}
